composables/useModal.ts

```diff
@@ -1,49 +1,46 @@
 import type { Modal } from '@/types/modal'
 
 /**
- * Trigger modal
+ * Show modal
  *
  * @param data
  */
-const showModal = (modal: Modal | string) => {
-  const appStore = useAppStore()
+const showModal = async (modal: Modal) => {
+  const route = useRoute()
+  const { isModalOpen } = storeToRefs(useAppStore())
 
-  if (appStore.modal?.isOpen) {
+  if (isModalOpen.value) {
     return
   }
 
-  if (typeof modal === 'object' && modal.isUrlModal) {
-    return setModalQueryParams(modal)
-  }
-
-  const modalDefaults: Modal = {
-    isOpen: true,
-    icon: '/images/up-error.png',
-    size: 'small',
+  const modalQueryParams: ModalQueryParams = {
+    modalTemplate: modal.template || 'Default',
+    modalSize: modal.size || 'small',
+    modalData: JSON.stringify(modal.data),
   }
 
-  if (typeof modal === 'string') {
-    appStore.setModal({ ...modalDefaults, message: modal })
-  } else {
-    appStore.setModal({ ...modalDefaults, ...modal })
-  }
+  await navigateTo({
```

Comment:
Not sure I understand the idea of navigate here. So on opening/closing modal you navigate?

Reply:
`showModal` / `hideModal` just change query params. Later in `AppModal.vue` there is a watcher that check for params and fill component with right template and data.

Reply:
Ok I never seen a modal changing route params that's why but ok I understand now

Reply:
It's quite innovative solution I agree 😅 Fabs wanted to be able to re-open modals with browser back button.